extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugPhotosExporter.java

```diff
@@ -93,48 +89,54 @@ public SmugMugPhotosExporter(
     IdOnlyContainerResource resource =
         (IdOnlyContainerResource) exportInformation.getContainerResource();
 
+    SmugMugInterface smugMugInterface;
+
+    try {
+      smugMugInterface = getOrCreateSmugMugInterface(authData);
```

Comment:
can smugmugInterface be initialized in the constructor? can we use the member var instead of passing it to the export methods?

Reply:
Ideally yes, esp if each job/worker will only handle one data transfer and then exit. But the issue right now is that they are reused (and the extensions don't provide a brand new instance each time getExporter() or getImporter() is called). Each service instance is specific to the authdata passed in and makes requests specifically for that user. 

This is the same pattern that is done in the Google services. For Flickr, we just call a function to set the request context to the new user auth data. This seemed better so that we don't accidentally forget to reset the context.

Reply:
Yeah, the auth data is request time (and should be as we may need to refresh expired tokens) but some sdk's and client require it at construction time. There may be better refactoring here, e.g. use a factory to wrap this functionality or some other pattern but we figured we'd do that once we get all the code migrated.